src/main/java/frc/robot/Subsystem/Shooter/Shooter.java

```diff
@@ -0,0 +1,141 @@
+    slot0Configs.kD = UP_KD; // no output for error derivative
+
+    // set Motion Magic Velocity settings
+    var motionMagicConfigs = talonFXConfigs.MotionMagic;
+    motionMagicConfigs.MotionMagicAcceleration = ACCELERATION; // Target acceleration of 400 rps/s (0.25 seconds to max)
+    motionMagicConfigs.MotionMagicJerk = JERK; // Target jerk of 4000 rps/s/s (0.1 seconds)
+
+    //Limits
+    var limitConfigs = new CurrentLimitsConfigs();
+    // enable stator current limit
+    limitConfigs.StatorCurrentLimit = CURRENT_LIMIT;
+    limitConfigs.StatorCurrentLimitEnable = true;
+
+    //upload configs to motor
+    StatusCode statusCode = StatusCode.StatusCodeNotInitialized;
+    for (int i = 0; i < 5; i++){
+      statusCode = m_upMotor.getConfigurator().apply(talonFXConfigs);
+      if (statusCode.isOK())
+        break;
+    }
+    if (!statusCode.isOK())
+      System.out.println("Shooter Arm could not apply config, error code:" + statusCode.toString());
+  }
+  private void downConfigs(){
+    var talonFXConfigs = new TalonFXConfiguration();
+    // set slot 0 gains
+    var slot0Configs = talonFXConfigs.Slot0;
+    slot0Configs.kS = DOWN_KS; // Add 0.25 V output to overcome static friction
+    slot0Configs.kA = DOWN_KA; // An acceleration of 1 rps/s requires 0.01 V output
+    slot0Configs.kP = DOWN_KP; // An error of 1 rps results in 0.11 V output
+    slot0Configs.kI = DOWN_KI; // no output for integrated error
+    slot0Configs.kD = DOWN_KD; // no output for error derivative
+
+    // set Motion Magic Velocity settings
+    var motionMagicConfigs = talonFXConfigs.MotionMagic;
+    motionMagicConfigs.MotionMagicAcceleration = ACCELERATION; // Target acceleration of 400 rps/s (0.25 seconds to max)
+    motionMagicConfigs.MotionMagicJerk = JERK; // Target jerk of 4000 rps/s/s (0.1 seconds)
+
+    //Limits
+    var limitConfigs = new CurrentLimitsConfigs();
+    // enable stator current limit
+    limitConfigs.StatorCurrentLimit = CURRENT_LIMIT;
+    limitConfigs.StatorCurrentLimitEnable = true;
+
+    //upload configs to motor
+    StatusCode statusCode = StatusCode.StatusCodeNotInitialized;
+    for (int i = 0; i < 5; i++){
+      statusCode = m_downMotor.getConfigurator().apply(talonFXConfigs);
+      if (statusCode.isOK())
+        break;
+    }
+    if (!statusCode.isOK())
+      System.out.println("Shooter Arm could not apply config, error code:" + statusCode.toString());
+ }
+    /**
+     * set the up motor at a given speed 
+     * @param speed
+     */
+    public void setSpeedUp(double speed){
+        m_upMotor.setControl(mm.withVelocity(speed));
+    }
+    /**
+     * set the down motor at a given speed 
+     * @param speed
+     */
+    public void setSpeedDown(double speed){
+        m_downMotor.setControl(mm.withVelocity(speed));
+    }
+    /**
+     * set both motors at a given speed 
+     * @param speed
+     * @return runOnce Command
+     */
+    public Command shoot(double speed){
```

Comment:
it isn't telling the motors to get to SHOOT_SPEED so it isn't really shoot and more of set speed